src/phoenix/server/api/dataloaders/cache/two_tier_cache.py

```diff
@@ -0,0 +1,72 @@
+from abc import ABC
+from asyncio import Future
+from typing import Any, Callable, Generic, Optional, Tuple, TypeVar, cast
+
+from cachetools import Cache, LFUCache
+from strawberry.dataloader import AbstractCache
+
+_Key = TypeVar("_Key")
+_Result = TypeVar("_Result")
+
+_Section = TypeVar("_Section")
+_SubKey = TypeVar("_SubKey")
+
+
+class TwoTierCache(
+    AbstractCache[_Key, _Result],
+    Generic[_Key, _Result, _Section, _SubKey],
+    ABC,
```

Comment:
Not seeing any abstract methods. Are you trying to prevent this class from instantiated directly?

Reply:
yes